components/atom/switch/src/index.js

```diff
@@ -0,0 +1,158 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import {ToggleSwitchTypeRender} from './SwitchType/toggle'
+import {SingleSwitchTypeRender} from './SwitchType/single'
+
+export const BASE_CLASS = 'sui-AtomSwitch'
+
+export const SIZES = {
+  DEFAULT: 'default',
+  LARGE: 'large'
+}
+
+export const TYPES = {
+  TOGGLE: 'toggle',
+  SELECT: 'select',
+  SINGLE: 'single'
+}
+
+class AtomSwitch extends Component {
+  state = {
+    toggle: false,
+    isFocus: false
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.executeIfEnabledFocusSwitch = this.executeIfEnabled(this.focusSwitch)
+    this.executeIfEnabledFocusOutSwitch = this.executeIfEnabled(this.focusOutSwitch)
+    this.executeIfEnabledToggleSwitch = this.executeIfEnabled(this.toggleSwitch)
+    this.executeIfEnabledActivateSwitch = this.executeIfEnabled(this.activateToggle)
+    this.executeIfEnabledDeactivateSwitch = this.executeIfEnabled(this.deactivateToggle)
+  }
+
+  componentDidMount () {
+    window.addEventListener('keydown', this.keyBindings.bind(this), true)
+  }
+
+  componentWillUnmount () {
```

Comment:
This won't work. When you do bind, you are creating a new instances of Function. You need to provide the same listener instance to actually remove it.